src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConstructorInitTests.cs

```diff
@@ -607,5 +609,183 @@ public void BadExpressionConstructor()
                 //     dynamic d = F() * 2;
                 Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "F()").WithArguments("Microsoft.CSharp.RuntimeBinder.CSharpArgumentInfo", "Create").WithLocation(4, 17));
         }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_01()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static int i = 0;
+    static bool b = false;
+}";
+            CompileAndVerify(source).VerifyMemberInIL("C..cctor()", false);
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_02()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static string s = null!;
+}";
+            CompileAndVerify(source).VerifyMemberInIL("C..cctor()", false);
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_03()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static (int, object) pair = (0, null!);
+}";
+            CompileAndVerify(source).VerifyIL("C..cctor()", @"{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldc.i4.0
+  IL_0001:  ldnull
+  IL_0002:  newobj     ""System.ValueTuple<int, object>..ctor(int, object)""
+  IL_0007:  stsfld     ""System.ValueTuple<int, object> C.pair""
+  IL_000c:  ret
+}");
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_04()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static (int, object) pair1 = default;
+    static (int, object) pair2 = default((int, object));
+    static (int, object) pair3 = default!;
+    static (int, object) pair4 = default((int, object))!;
+}";
+            CompileAndVerify(source).VerifyMemberInIL("C..cctor()", false);
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_05()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static C instance = default!;
+}";
+            CompileAndVerify(source).VerifyMemberInIL("C..cctor()", false);
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_06()
+        {
+            string source = @"
+#nullable enable
+
+struct S
+{
+    public int x;
+    public int y;
+}
+
+class C
+{
+    static S field1 = default;
```

Comment:
>default [](start = 22, length = 7)

Please also test:
- ```new S()``` and similar scenarios for nullable value type.
- Initializing a nullable value type with ```null```